File: 2D_MMO_Server/DBServer/ServerSession.cpp
#include "pch.h"
#include "ServerSession.h"
#include "Manager.h"

void ServerSession::OnConnected()
{
	cout << "serversession onconnect\n";
	RegisterRecv();
}

void ServerSession::OnDisconnected()
{
}

void ServerSession::OnSend(int32 len)
{
	cout << "OnSend\n";
}

int32 ServerSession::OnRecvPacket(BYTE* buffer, int32 len)
{
	Manager::Packet.OnRecvPacket(this, buffer);

	return len;
}

File: 2D_MMO_Client/Assets/Scripts/Session/ServerSession.cs
using ServerCore;
using System;
using System.Net;
using UnityEngine;

public class ServerSession : PacketSession
{
    public override void OnConnect(EndPoint endPoint)
    {
        Debug.Log("OnConnect");
    }
    public override void OnDisconnect(EndPoint endPoint)
    {
        Debug.Log("OnDisconnect");
    }

    public override void OnRecvPacket(ArraySegment<byte> data)
    {
        Manager.Network.Push(data);
    }
    public override void OnSend(int numOfBytes)
    {
        //Debug.Log("OnSend");
    }
}
